### ISSUE TITLE
Sass-lint complaining about valid strictbem class name

### ISSUE THREAD

**Issue by c-emil** [sasstools/sass-lint#1036]
### What version of Sass Lint are you using?
1.10.2

### Please include any relevant parts of your configuration
```yml
rules:
  class-name-format:
    - 1
    - allow-leading-underscore: false
    - convention: strictbem
    - convention-explanation: 'For example: ...'
```

### What did you do? Please include the actual source code causing the issue.
If I write class like this `.block-name__element`, it gives me warnings. Writing just `.block-name` passes ok.

### What did you expect to happen?
If I follow strict BEM naming, it should not give me any warning. Also if I don't follow strict BEM, it should give my my custom convention-explanation, but instead, it gives me the following error.

### What actually happened? Please include any error messages given to you by Sass Lint.
```bash
warning  Class '.block-name__element' should be written in lowercase with hyphens
```

### If you're using a IDE plugin have you tried the CLI too?
I'm using Atom, however I've tried CLI as well and it gives me same warning.

**Comment by matter123** [comment]
You need to make all of the configuration options under one key.
ex:
```yaml
rules:
  class-name-format:
    - 1
    -
      allow-leading-underscore: false
      convention: strictbem
      convention-explanation: 'For example: ...'
```
Sass-lint expects optional arguments to all be in one configuration object

**Comment by c-emil** [comment]
@matter123 Thanks very much!

**Comment by maikeldaloo** [comment]
They really need to make this obvious in their docs. 

Thank you so much for this. Very helpful.
